sycl/test/basic_tests/builtins/nextafter.cpp

```diff
@@ -0,0 +1,88 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %t.out
+
+#include <sycl/sycl.hpp>
+#include <cassert>
+#include <cmath>
+#include <limits>
+
+template<typename T>
+struct nextafter_test {
+  T* deviceInputs;
+  T* deviceOutputs;
+  size_t numValues;
+  void operator()() const {
+    for (size_t i{0}; i<numValues; ++i) {
+      deviceOutputs[i] = sycl::nextafter(deviceInputs[2*i], deviceInputs[2*i+1]);
+    }
+  }
+};
+
+struct expect {
+  double d;
+  float f;
+  sycl::half h;
+  template<typename T>
+  T get() {
+    if constexpr (std::is_same_v<T,double>) return d;
+    if constexpr (std::is_same_v<T,float>) return f;
+    if constexpr (std::is_same_v<T,sycl::half>) return h;
+    return T{};
+  }
+};
+
+template<typename T>
+void run_nextafter_test(sycl::queue& deviceQueue) {
+  constexpr static size_t NumTests{11};
+  std::array<T,2*NumTests> inputs{
+     0.0,  0.0, // positive zeros
+    -0.0, -0.0, // negative zeros
+    -0.0,  0.0, // neg and pos zero
+     0.0, -0.0, // pos and neg zero
+     1.0,  2.0, // positive away from zero
+     1.0,  0.0, // positive towards zero
+    -1.0,  0.0, // negative towards zero
+    -1.0, -2.0, // negative away from zero
+    std::numeric_limits<T>::quiet_NaN(), 1.0, // NaN first arg
+    1.0, std::numeric_limits<T>::quiet_NaN(), // NaN second arg
+    std::numeric_limits<T>::quiet_NaN(), std::numeric_limits<T>::quiet_NaN(), // NaN both args
+  };
+  std::array<T,NumTests> outputs{};
+  std::array<T,NumTests> expected{
+    0.0,
+    -0.0,
+    0.0,
+    -0.0,
+    expect{1.0000000000000002, 1.0000001f, 1.001f}.get<T>(),
+    expect{0.99999999999999988897769753748434595763683319091796875, 0.999999940395355224609375f, 0.99951171875f}.get<T>(),
+    expect{-0.99999999999999988897769753748434595763683319091796875, -0.999999940395355224609375f, -0.99951171875f}.get<T>(),
+    expect{-1.0000000000000002, -1.0000001f, -1.001f}.get<T>(),
+    expect{sycl::nan((unsigned long){0}), sycl::nan((unsigned int){0}), sycl::nan((unsigned short){0})}.get<T>(),
+    expect{sycl::nan((unsigned long){0}), sycl::nan((unsigned int){0}), sycl::nan((unsigned short){0})}.get<T>(),
+    expect{sycl::nan((unsigned long){0}), sycl::nan((unsigned int){0}), sycl::nan((unsigned short){0})}.get<T>(),
+  };
+
+  T* deviceInputs{sycl::malloc_device<T>(2*NumTests, deviceQueue)};
+  T* deviceOutputs{sycl::malloc_device<T>(NumTests, deviceQueue)};
+  deviceQueue.copy(inputs.data(), deviceInputs, 2*NumTests).wait_and_throw();
+  deviceQueue.submit([deviceInputs,deviceOutputs](sycl::handler& cgh){
+    cgh.single_task(nextafter_test<T>{deviceInputs,deviceOutputs,NumTests});
+  }).wait_and_throw();
```

Comment:
Since this needs a device, the test should be in `sycl/test-e2e`. In general `sycl/test` either needs to be compile-only tests or be tests that can run without a device being present.

Reply:
Thanks for explaining! Moved to `sycl/test-e2e/Basic/built-ins`